lib/matplotlib/backends/qt_editor/figureoptions.py

```diff
@@ -165,37 +177,35 @@ def prepare_data(d, init):
 
     def apply_callback(data):
         """A callback to apply changes."""
-        orig_xlim = axes.get_xlim()
-        orig_ylim = axes.get_ylim()
+        orig_limits = {
+            axis: getattr(axes, f"get_{axis}lim")()
+            for axis in axis_map.keys()
+        }
 
         general = data.pop(0)
         curves = data.pop(0) if has_curve else []
         mappables = data.pop(0) if has_sm else []
         if data:
             raise ValueError("Unexpected field")
 
-        # Set / General
-        (title, xmin, xmax, xlabel, xscale, ymin, ymax, ylabel, yscale,
-         generate_legend) = general
+        title = general.pop(0)
+        axes.set_title(title)
+        generate_legend = general.pop()
 
-        if axes.get_xscale() != xscale:
-            axes.set_xscale(xscale)
-        if axes.get_yscale() != yscale:
-            axes.set_yscale(yscale)
+        for i, axis in enumerate(axis_map.keys()):
+            ax = getattr(axes, f"{axis}axis")
+            axmin = general[4*i]
```

Comment:
Could one `pop` here instead?

Reply:
I used indexing instead of `pop(0)` because `pop` has shown to be considerably less performant on lists when used to pop only the first element whereas indexing would be O(1). It may not impact a lot in this usecase right away but that was my reasoning. If you think `pop` fits better with the code style then I'll change it.

Reply:
Sure! Keep it like this then.